pkg/kubelet/pod_workers.go

```diff
@@ -625,6 +632,15 @@ func (p *podWorkers) CouldHaveRunningContainers(uid types.UID) bool {
 	return !p.podsSynced
 }
 
+func (p *podWorkers) IsPodTerminationFinished(uid types.UID) bool {
+	p.podLock.Lock()
+	defer p.podLock.Unlock()
+	if status, ok := p.podSyncStatuses[uid]; ok {
+		return status.IsFinished()
+	}
+	return false
```

Comment:
If we do decide to implement this method (because we can't justify collapsing teardown), then we have to decide whether this is false or depends on the synced status.  Generally the teardown methods (things that gate other components from completing cleanup) depend on sync status - we can't safely tear something down if we aren't certain whether it should still be present.  I.e. this is a subtle but important part of the definition.  Teardown methods are named `ShouldXXXX()`.  The `Is*` methods (the ones that return false if there is no sync status) are usually only safe to call from inside the pod worker (because they know for certain the pod UID exists in status at that time).

Since this is called from status manager, I would generally consider it a teardown method and so it must depend on the sync status to return true (and the name would be wrong).

Reply:
I see, I suggest a new name `CouldHavePendingSyncTerminatedPod` which directly reflects the intention - we need to maintain the status in cache until the SyncTerminatedPod is completed (or it is the request itself). It is implemented by analogy to `CouldHaveRunningContainers`. PTAL.

Reply:
Another name that comes to my mind is `ShouldHaveSyncTerminatedPodFinished`.

Reply:
I suggested `ShouldPodBeFinished` instead - shorter and matches the term we use inside the pod worker.

Reply:
Done